assignments/lambda-classes.js

```diff
@@ -1 +1,178 @@
+
+  class Instructor extends Person {
+      constructor(uniqueProps){
+          super(uniqueProps);
+          this.specialty = uniqueProps.specialty;
+          this.favLanguage = uniqueProps.favLanguage;
+          this.catchPhrase = uniqueProps.catchPhrase;
+      }
+      demo(subject) {
+          return `Today we are learning about ${subject}`
+      }
+      grade(Student, subject) {
+          return `${Student.name} receives a perfect score on ${subject}`
+      }
+      popQuiz(Student) {
+        let points = Math.random();
+        if (points > .5) {
+            return Student.grade += (points * 10);  
+        } else if (points <= .5) {
+            return Student.grade -= (points * 10);  
+        }
+      }
+  }
+
+
+  const chang = new Instructor({
+    name: 'Chang',
+    location: 'Greendale',
+    age: 37,
+    gender: 'male',
+    favLanguage: 'JavaScript',
+    specialty: 'Front-end',
+    catchPhrase: `I'll allow it`
+  });
+
+
+  const fred = new Instructor({
+    name: 'Fred',
+    location: 'Bedrock',
+    age: 37,
+    gender: 'male',
+    favLanguage: 'JavaScript',
+    specialty: 'Front-end',
+    catchPhrase: `Don't forget the homies`
+  });
+
+  console.log(fred.speak())
+  console.log(chang.speak())
+
+
+  class Student extends Person {
+      constructor(studentProps) {
+          super(studentProps);
+          this.previousBackground = studentProps.previousBackground;
+          this.className = studentProps.className;
+          this.favSubjects = studentProps.favSubjects;
+          this.grade = studentProps.grade;
+      }
+        listsSubjects() {
+          return this.favSubjects;
+      }
+        PRAssignment(subject) {
+          return `${this.name} has submitted a PR for ${subject}`
+      }
+        sprintChallenge(subject) {
+        return `${this.name} has begun sprint challenge on ${subject}`
+      }
+        graduate () {
+        if (this.grade >= 70) {
+            return `Congrats, ${this.name}, you graduate!`
+        } else if (this.grade < 70) {
+            return `Sorry, ${this.name}, You don't graduate!`
+        }
+    }
+  }
+
+  const troy = new Student({
+    name: 'Troy',
+    location: 'Greendale',
+    age: 21,
+    gender: 'male',
+    previousBackground: 'football',
+    className: 'CS132',
+    favSubjects: ['Html', 'CSS', 'JavaScript'],
+    grade: 60,
+  });
+
+  const abed = new Student({
+    name: 'Abed',
+    location: 'Greendale',
+    age: 20,
+    gender: 'male',
+    previousBackground: 'Asbergers',
+    className: 'CS131',
+    favSubjects: ['Ruby', 'Videography', 'JavaScript'],
+    grade: 90
+  });
+
+
+  class ProjectManagers extends Instructor {
+    constructor(pmProps) {
+        super(pmProps);
+        this.gradClassName = pmProps.gradClassName;
+        this.favInstructor = pmProps.favInstructor;
+    }
+    standup (channel) {
+        return `${this.name} announces to ${channel}, @channel stand up time!`
+    }
+    debugsCode (Student, subject) {
```

Comment:
I noticed that you passed in Student.  I assume this is because it said to pass in a student object,  but remember this is just the argument passed in.  student would have worked.